### DISCUSSION TITLE
Wie kann man Routen in Cruiser umdrehen?

### DISCUSSION THREAD

**Discussion by RolfGS** [devemux86/cruiser#424] [Q&A] [answered]
Hallo,
ich bin neu hier in GitHub und hoffe nichts zu falsch zumachen- bitte um Nachsicht:
Seit 2 Tagen bin ich Cruiser Abonnent bzw seit 2 Wochen Kurviger3 Abonnent und teste gerade die Kurviger/Cruiser Kombination am Smartphone im Auto (weil Motorradfahren mom nicht geht).
Torsten und Wolfgang haben mir sehr komeptent geholfen das alles schnell in Gang zu bekommen - vielen lieben Dank an dieser Stelle!
Nun meine Frage: ich habe gestern im Auto die Cruiser Navi gestartet und festgestellt dass meine aus Google Drive geladene Route
(zuvor in Kurviger geplant und als .kurviger Datei in Google Drive exportiert) verkehrt rum routet. 
Nun hätte ich die Route am liebsten in Cruiser "umgekehrt".

Ist das möglich ? 
Oder muss ich erst wieder die Route in Kurvige laden, umkehren und in den Google Drive stellen und dann wieder in Cruiser importieren u. Navi neu starten - was halt recht umständlich ist wenn man schon im Auto/aufm Moped sitzt.

Gruss
Rolf Reitenbach München

**Comment by devemux86** [comment, marked as answer]
Hi @RolfGS,

To reverse a route, long press on the route line and use the menu as shown in the pictures.
A reverse route is calculated with the selected routing service, such as BRouter (offline).
If you have any further questions, please let us know!

Um eine Route umzukehren, drücken Sie lange auf die Routenzeile und verwenden Sie das Menü wie in den Bildern.
Es wird eine umgekehrte Route mit dem ausgewählten Routing-Dienst, wie BRouter (offline), berechnet.
Wenn Sie weitere Fragen haben, lassen Sie es uns bitte wissen!

<img src="https://github.com/devemux86/cruiser/assets/3484020/e4f8abae-0ae8-43b1-ac06-921439aa3545" width="50%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/226c39e6-acfb-44b9-827c-d7650d98a4a8" width="50%"/>

**Reply by RolfGS** [reply]
Hi, thanks this works - and thanks to your instant answer - Torsten already praised your feedback.

Obseravtions:
First I loaded the route under OpenRouteService and when I inverted the route I had all of sudden a "stich street" (Stichstrasse) in my route. Loading the Kurviger route under OpenRouteService e.g. normally did not create that stich street.
When I changed to BRouter (offline), the inverting was calculated without stich street, also the normal loading from Kurviger/Google Drive.

Would you recommend to always use BRouter offline? ...excuse, I am not very familiar with such details...

beszt regards,
Rolf

**Reply by WolfgangZ42** [reply]
Hi Rolf,
for my opinion BRouter (offline) is the right choise - I did a litte test with Graphhopper as well, but now I'll stick to BRouter (offline). 
BR Wolfgang

**Reply by devemux86** [reply]
@RolfGS Routing services work differently, each has its own routing options.

OpenRouteService and Valhalla are good fast routers with many options, but they are online.

We recommend trying BRouter (offline), now included in Cruiser for easy use and is **offline**.
It has many routing options and can even adjust the strength of the avoidances.

Note: you can write in any language you are comfortable. :slightly_smiling_face: 

<img src="https://github.com/devemux86/cruiser/assets/3484020/c341cb60-90df-4547-963e-ce355bfb0844" width="50%"/>

**Comment by RolfGS** [comment]
Ok, you convinced me, I will use BRouter Offline...
Just started BRouter card download (Germany, Austria,Italy, Greecce, some France...) hope this was not wrong...

One more question: if I copy the Kurviger routes (that I created at PC and loaded on Google Drive) on my smartphone, is there a possiblility 
in Cruiser to start navigation form routes on smartphone (= import routes not from Cloud(/Google Drive, but from a file on my smartphone)? I am asking because as tourguide it may be helpful to import a route if no access to internet/cloud is possible.

**Reply by devemux86** [reply]
@RolfGS 

> if I copy the Kurviger routes (that I created at PC and loaded on Google Drive) on my smartphone, is there a possiblility
in Cruiser to start navigation form routes on smartphone (= import routes not from Cloud(/Google Drive, but from a file on my smartphone)?

Cruiser can import Kurviger routes _offline without internet_. So you can store them locally as files in your device.

Even better you can store them inside Cruiser as _favorite routes_ for easier access:

<img src="https://github.com/devemux86/cruiser/assets/3484020/95152bf5-dac6-492d-a99e-4feae8d3c5ab" width="50%"/>

**Comment by RolfGS** [comment]
Very nice, I just down loaded a couple routes in a new folder "Tuscany" (-> Meien Orte-->Routen-->Toskana) and this works very intuitively!
This is what I need in my private tourguide function I think.

**Comment by RolfGS** [comment]
<img width="182" alt="image" src="https://github.com/devemux86/cruiser/assets/156905078/37cb0fbb-b8c8-454d-b782-2bb67860a8b9">
<img width="164" alt="image" src="https://github.com/devemux86/cruiser/assets/156905078/91582fdb-fde6-4873-a3ff-8f309ca4ba69">

Hello,
with BRouter offline I still get stich streets (Stichstrassen see picture 2,  point 1 and 4) after inverting.

**Reply by devemux86** [reply]
@RolfGS is the left image an imported kurviger route?

It depends on what options the kurviger route had and what BRouter options you have selected in Cruiser.
Have you selected any avoid options in BRouter settings?

Can you attach the route file here so we can check it?

**Comment by RolfGS** [comment]
Hallo, hier meine BRouter Einstellungen:
<img src="https://github.com/devemux86/cruiser/assets/156905078/52f4f78f-f41e-411d-b79a-33343301c6f8" width="20%"/><img src="https://github.com/devemux86/cruiser/assets/156905078/2743996b-b2cb-4b33-b3e1-10fc6d3821f9" width="20%"/>
<img src="https://github.com/devemux86/cruiser/assets/156905078/d483a86d-ff51-48de-8f8c-dcf2b4930753" width="20%"/><img src="https://github.com/devemux86/cruiser/assets/156905078/149b4daa-0bae-4c95-81dc-d8b9a5542750" width="20%"/>

**Reply by devemux86** [reply]
@RolfGS 

You use the default BRouter settings without activated avoidance options, e.g. highways can be used.
If your kurviger route had avoidance options, you must activate similar BRouter options.

Sie verwenden die Standard-BRouter-Einstellungen ohne aktivierte Vermeidungsoptionen, z.B. können Autobahnen benutzt werden.
Wenn Ihre kurviger Route Vermeidungsoptionen hatte, müssen Sie ähnliche BRouter-Optionen aktivieren.

**Comment by RolfGS** [comment]
Sorry dass sie so gross herauskamen...

**Reply by devemux86** [reply]
@RolfGS 

No problem, here is a way to change the size of images in GitHub:

Kein Problem, hier eine Möglichkeit, die Größe von Bildern in GitHub zu ändern:

- https://github.com/devemux86/cruiser/discussions/27

**Comment by RolfGS** [comment]
<img width="334" alt="image" src="https://github.com/devemux86/cruiser/assets/156905078/b7754cc0-25a7-4ae8-8020-0629e4e240f8">
Hello, I checked my Kurviger settings (picture) and they look like the BRouter Default settings.
Do I overlook sth at Kurviger settings ?

**Reply by devemux86** [reply]
@RolfGS what route type do you use in Kurviger?
Fastest or fast and curvy or curvy or extra curvy route?

**Reply by RolfGS** [reply]
Usually I use extra curvy

**Reply by devemux86** [reply]
@RolfGS 

According to their documentation for extra curvy: "The calculated routes are often small country roads with little traffic, but they usually involve longer detours".

BRouter will use the fastest route and motorways if we don't enable avoid options.

To have a curvy routing in BRouter you need to enable the avoid options for major roads, like motorways, primary roads, etc.
It's recommended that you play around with the BRouter options until you find what works best for your routing preferences.

**Reply by WolfgangZ42** [reply]
>  I checked my Kurviger settings (picture) 

@RolfGS 
Not the "Vermeidungsstärken" (avoidance strength) are relevant but the "Vermeidungen" (avoidances) and for sure the curviness:
![2024-01-18 09_13_18-Achenkirch - Brandenberg _ Kurviger](https://github.com/devemux86/cruiser/assets/121806238/69920de5-1755-4764-8c1d-84675fc0532d)
In this example the avoidance strength is totally irrelevant since no avoidance option is selected.

**Reply by RolfGS** [reply]
My settings are exactly the same with exception that I selected very curvy..maybe this is the problem? I will follow emux suggestions and play around a littel bit with the BRouter settings. I assume that if I want to change my Kurviger settings from very to medium curvy then I need to reload/recalculate all routes and export then again in Kurviger for Cruiser useage.

**Reply by RolfGS** [reply]
Hello it seems to work, I set mototways and primary roads to 100% and inverting worked at 3 routes without any stich streets. Think I am set for time being and will do further tests in car and when warmer on the bike.
Thx again !

**Reply by devemux86** [reply]
@RolfGS 

Note that 100% (red color) is an absolute avoid option, it will never use this type of road.

Maybe you could also try a step lower to 80%.

**Comment by RolfGS** [comment]
<img width="701" alt="image" src="https://github.com/devemux86/cruiser/assets/156905078/7354d31a-38b6-4fac-bf3f-dc54a53d6000">
Think I understood how resize...I went into "Bearbeiten" in upper right corner of message and then I changed the % in each line from 50%->20%, e.g..

**Comment by RolfGS** [comment]
To Emux: I will play around a little with the setting as you suggest...thansk for both of you for your help!

**Reply by devemux86** [reply]
@RolfGS 

The difference is that Kurviger has some modes (curvy, extra curvy) that internally / automatically avoid road types.

For BRouter in Cruiser we can enable avoid options for many road types and strengths.

---

There is a discussion about introducing automatic curvy modes in BRouter:

- https://github.com/devemux86/cruiser/discussions/254

**Comment by RolfGS** [comment]
will do!

Holen Sie sich Outlook für Android<https://aka.ms/ghei36>
________________________________

**Comment by RolfGS** [comment]
it did not work...will try later once more...

Holen Sie sich Outlook für Android<https://aka.ms/ghei36>
________________________________

**Reply by devemux86** [reply]
@RolfGS It may depend on the specific route or area and the available roads.

It is worth experimenting with many options and values.

**Comment by RolfGS** [comment]
hello, imberting only works with highways and mainroads at 100%, tried 60, 80, 100% in all combinations.

Holen Sie sich Outlook für Android<https://aka.ms/ghei36>
________________________________

**Reply by RolfGS** [reply]
On a note to this previous comment of myself: these "no inverting problems with 100% settings for highways and mainroads" only applies for roundtrips I generated automatically in Kurviger and loaded them into my routes in Cruiser on my smartphone. I spent no energy in finetuning certain curves with additional shaping points etc.. As a lazy person I appreciate the easy and intuitive roundtrip feature in Kurviger as this motivates me to more upfront planning and playing around  with these tools. But the more I will "play" with all of that, the more I will get in the areas of planning that you are familiar with. I just changed from BMW Nav5/Basecamp scrap to Kurviger/Cruiser/WunderLINQ smartphone navigation and am in a steep learning curve to prepare myself for 1st test rides as ossn as weather will allow it in order to get safe until I do my private Tour guide job in June in Tuscany. Cruiser/Kurviger etc setup motivates me to dive deeper in route planning because with Basecamp I always got frustrated.

**Reply by devemux86** [reply]
@RolfGS 

If you have questions or ideas for improvements and new features, please write them and we will discuss them.

Cruiser is constantly improving and getting new features with every update. :slightly_smiling_face:

**Reply by RolfGS** [reply]
Will do, thanks!

**Reply by MotoUKRider** [reply]
@RolfGS For info, I use BaseCamp (and Freizeitkarte maps) for all my route planning on the PC, export to GPX and open the routes directly in Cruiser (which obeys the Shaping-Points *and* Waypoints perfectly, ie: it gives me time+distance to each "cake stop" :cake:).  Lots of people give BaseCamp a hard time, but I like its filing system (tree roots, lists, duplicate Waypoints in different routes etc) and I can flick between routes without recalculation instantly.  My route planning is now at the stage where any route planned on the PC looks identical in Cruiser on my phone (I have "Unpaved" as my only avoidance for normal riding).

I am more of a tourer than a "send me 50 clicks north with 75% curves" kinda rider, so I've got a method that works for me - and I'm happy to manually export GPX files, copy them to my phone via a cable and not have cloud sync or rely on a 3rd-party server to handle my up/downloads :smiley:

Cruiser Desktop is great for looking at roads in a clear/simple form, but I am not keen on the manual GPX load/save (being spoiled with BaseCamp's instant folder view!).  Where it does help is when I load a route into it - I can easily see if there are any existing Favorites along it, displaying them *all* in BaseCamp isn't easy unless you get messy with list contents.  If I spot a good place on the Cruiser map, I can find it in BaseCamp and put a copy into that Route List.

If the Desktop app was to develop further, an Outlook "inbox" style filing system would probably get me to convert :laughing: 

Ask 5 people how they do it and you'll get 5 different answers - I'm sure my way seems like a big chore to some, but I'd much rather blame myself for a forgotten GPX than be stuck without data on my phone waiting for a website to refresh or a map portion to render!

**Comment by MotoUKRider** [comment]
I'll echo the suggestions here: A Sat-Nav, in its default configuration will get you from A to B as quickly/efficiently as possible.  If you have a Shaping-Point/Waypoint in the middle of a "fun" road that also runs parallel to a boring highway, it will likely put you on the highway until it reaches the nearest junction, sending you to that "fun" marker and then back out to the highway to continue your boring journey(!)

More Shaping-Points or fine-tuning of the Avoidance levels (create a spare profile to tweak, call it "Curvy" or whatever).  I always add Shaping-Points to my routes when there's a nice road near a boring motorway :smile:   I could also set my "Max Speed" to 65mph meaning that I should never be taken on a dual-carriageway (limit is generally 70mph).

"Curvy" or "Twisty" (in Calimoto or Kurviger) is just a set of Avoidance levels, which we have control of in BRouter - I suppose the easiest way to get a like-for-like route in Cruiser is to load the route and then gradually increase the avoidances for the largers roads until the route layout starts to look like your original plan.

YMMV, as they say :hand:

**Reply by devemux86** [reply]
@MotoUKRider thanks, very nice description! :+1:

**Reply by RolfGS** [reply]
Hello and also thanks for that nice description. I will consider the Shaping points increase in nice roads and the max speed trick in my future route plannings. For the moment I have 10 Tuscany routes already planned in Kurviger and loaded in my Cruiser file in smartphone, these I do not want to touch anymore (but as they are to and back style routes in Italian countryside I do not expect any problems by inverting - which I may not di at all). But for roundtrips - and I love the easy Kurviger roundtrip planning - in future your suggestions make big sense.

**Comment by RolfGS** [comment]
Fair point. I am glad that Basecamp works perfectly for you. Since 2015 I never succeeded with Basecamp route planning and I am now convinced " Basecamp does not like me 🙂 - and vice versa. Friends of mine booked full day Basecamp planning sessions and now are possibly where you are. Every year they helped me to correct tracks and had explanations why Basecamp lost navigation when we left the track etc. . Since a friend showed me Kurviger it was like every Basecamp problem was gone right away for me. The Kurviger/ Cruiser combination with all routes loaded on my handy in a Cruiser file for me seems be now the best solution. I agree with you that I do not want depend on Internet connection while riding.
So now lets hope the snow gets away  soon that we can go back from pc planning to riding...wish you a great and safe riding year !

Holen Sie sich Outlook für Android<https://aka.ms/ghei36>
________________________________
